testing/cloud_layouts/wait-master-ssh-and-update-comment.sh

```diff
@@ -0,0 +1,205 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+log_file="$1"
+comment_url="$2"
+connection_str_out_file="$3"
+
+if [ -z "$log_file" ]; then
+  echo "Log file is required"
+  exit 1
+fi
+
+if [ -z "$comment_url" ]; then
+  echo "Comment url is required"
+  exit 1
+fi
+
+if [ -z "$TOKEN_GITHUB_BOT" ]; then
+  echo "Token env is required"
+  exit 1
+fi
+
+if [ -z "$connection_str_out_file" ]; then
+  echo "Connection string output file is required"
+  exit 1
+fi
+
+master_ip=""
+master_user=""
+result_body=""
+
+function get_comment(){
+  local response_file="$1"
+  local exit_code
+  local http_code
+  http_code="$(curl \
+    --output "$response_file" \
+    --write-out "%{http_code}" \
+    -H "Accept: application/vnd.github+json" \
+    -H "Authorization: Bearer $TOKEN_GITHUB_BOT" \
+    "$comment_url"
+  )"
+  exit_code="$?"
+
+  echo "Getting response (code: $http_code):"
+  cat "$response_file"
+
+  if [[ "$exit_code" != 0 ]]; then
+    echo "Incorrect response code $exit_code"
+    return 1
+  fi
+
+  if [[ "$http_code" != "200" ]]; then
+    echo "Incorrect response code $http_code"
+    return 1
+  fi
+
+  local connection_str="${master_user}@${master_ip}"
+  local connection_str_body="${PROVIDER}-${LAYOUT}-${CRI}-${KUBERNETES_VERSION} - Connection string: \`ssh ${connection_str}\`"
+  local bbody
+  if ! bbody="$(cat "$response_file" | jq -crM --arg a "$connection_str_body" '{body: (.body + "\r\n\r\n" + $a + "\r\n")}')"; then
+    return 1
+  fi
+
+  result_body="$bbody"
+  echo "Result body: $result_body"
+}
+
+function update_comment(){
+  local http_body="$1"
+  local response_file=$(mktemp)
+  local exit_code
+  local http_code
+
+  http_code="$(curl \
+    -v --output "$response_file" \
+    --write-out "%{http_code}" \
+    -X PATCH \
+    -H "Accept: application/vnd.github+json" \
+    -H "Authorization: Bearer $TOKEN_GITHUB_BOT" \
+    -d "$http_body" \
+    "$comment_url"
+  )"
+  exit_code="$?"
+
+  echo "Response update output:"
+  cat "$response_file"
+  rm -f "$response_file"
+
+  if [ "$exit_code" == 0 ]; then
+    if [ "$http_code" == "200" ]; then
+        return 0
+    fi
+  fi
+
+  echo "Comment not updated, http code: $http_code"
+
+  return 1
+}
+
+function wait_master_host_connection_string() {
+  local ip
+  if ! ip="$(grep -Po '(?<=master_ip_address_for_ssh = ).+$' "$log_file")"; then
+    echo "Master ip not found"
+    return 1
+  fi
+
+  #https://stackoverflow.com/posts/36760050/revisions
+  # we need to verify ip because string ca fsynced partially
+  if ! echo "$ip" | grep -Po '((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}'; then
```

Comment:
```suggestion
  # IP validation regex from https://stackoverflow.com/posts/36760050/revisions
  # IP should be verified because streaming log can contains partial string.
  if ! echo "$ip" | grep -Po '((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}'; then
```